runtime/codert_vm/cnathelp.cpp

```diff
@@ -1616,12 +1616,16 @@ slow_jitMonitorEnterImpl(J9VMThread *currentThread, bool forMethod)
 				resolveFrame->specialFrameFlags = (resolveFrame->specialFrameFlags & ~J9_STACK_FLAGS_JIT_FRAME_SUB_TYPE_MASK) | J9_STACK_FLAGS_JIT_FAILED_METHOD_MONITOR_ENTER_RESOLVE;
 			}
 		}
-#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
```

Comment:
you can keep the original macro guards, but instead of the VT check it can be a JDK16+ check.

Reply:
If it is changed to a JDK16+ check, does it affect the current internal Valhalla builds that is JDK15 and lower ?

Reply:
VT builds should be running on JDKnext. I believe that is currently JDK16 @JasonFengJ9 can you confirm

Reply:
@tajila that's correct, `Pipeline_Build_Test_JDKnext_x86-64_linux_valhalla` is built on OpenJDK head stream `git@github.com:ibmruntimes/openj9-openjdk-jdk.git` which is called `JDKnext` and currently is `JDK16`.

Reply:
Changed to use `JAVA_SPEC_VERSION >= 16`

Reply:
Changed to `JAVA_SPEC_VERSION >= 16`.